be/src/olap/memory/column_writer.h

```diff
@@ -0,0 +1,81 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "olap/memory/common.h"
+
+namespace doris {
+namespace memory {
+
+class Column;
+
+// Exclusive writer for column, also with some reader's functionality
+//
+// Example writer usage:
+//   scoped_refptr<Column> column;
+//   std::unique_ptr<ColumnWriter> writer;
+//   // create writer
+//   RETURN_IF_ERROR(column->write(&writer).ok());
+//   writer->insert(new_rowid, &value, 0);
+//   writer->update(rowid, &value, 0);
+//   writer->finalize(new_version);
+//   // get new column refptr
+//   // if a COW has been done, column will point to new column object
+//   // else column will remain the same.
+//   writer->get_new_column(&column);
+class ColumnWriter {
```

Comment:
Is the class a writer of class or a writer of part of column?
In my opinion, column will contain all data for a table "column", and its data is not written at one time. One Writer may update some part of column data.
If my understanding is right, I think ColumnBlockWriter/ColumnChunkWriter is a better name.

Reply:
> Is the class a writer of class or a writer of part of column?
It is the writer of a whole column, it will write all the inserts/updates of this column of a writetx.